src/array.jl

```diff
@@ -365,13 +339,30 @@ function convert(::Type{CategoricalArray{T, N, R}}, A::CategoricalArray{S, N}) w
     refs = convert(Array{R, N}, A.refs)
     CategoricalArray{unwrap_catvaluetype(T), N}(refs, pool)
 end
+
 convert(::Type{CategoricalArray{T, N}}, A::CategoricalArray{S, N, R}) where {S, T, N, R} =
     convert(CategoricalArray{T, N, R}, A)
 convert(::Type{CategoricalArray{T}}, A::CategoricalArray{S, N, R}) where {S, T, N, R} =
     convert(CategoricalArray{T, N, R}, A)
 convert(::Type{CategoricalArray}, A::CategoricalArray{T, N, R}) where {T, N, R} =
     convert(CategoricalArray{T, N, R}, A)
 
+# resolve dispmatch ambiguities with AbstractVector/Matrix
+convert(::Type{CategoricalArray{T, 1}}, A::CategoricalVector{S, R}) where {S, T, R} =
```

Comment:
I'm surprised this is needed given that above we have `convert(::Type{CategoricalArray{T, N}}, A::CategoricalArray{S, N, R})`, where `N` should be equivalent to hardcoding `1` here. Maybe try adding `R <: Integer` in that method?

Same for the next two methods.

Reply:
These appear to be necessary; even with `R<:Integer` I get a method ambiguity unless these definitions are present. Here's the error:

```julia                                                                                                                                                                                                                                                                           
julia> str = "{\"id\":[1,2,3],\"var\":[\"x\",\"y\",\"z\"]}"

julia> readx = JSON3.read(str, MyCustomType)
ERROR: MethodError: convert(::Type{CategoricalArray{String,1,R,V,C,U} where U where C where V where R<:Integer}, ::CategoricalArray{Any,1,UInt32,Any,CategoricalValue{Any,UInt32},Missing}) is ambiguous. Candidates:
  convert(::Type{CategoricalArray{T,1,R,V,C,U} where U where C where V where R<:Integer}, A::AbstractArray{T,1} where T) where T in CategoricalArrays at /Users/davidlittle/Documents/work/tools/CategoricalArrays.jl/src/array.jl
:291       
  convert(::Type{CategoricalArray{T,N,R,V,C,U} where U where C where V where R<:Integer}, A::CategoricalArray{S,N,R,V,C,U} where U where C where V) where {S, T, N, R<:Integer} in CategoricalArrays at /Users/davidlittle/Documen
ts/work/tools/CategoricalArrays.jl/src/array.jl:344
Possible fix, define    
  convert(::Type{CategoricalArray{T,1,R,V,C,U} where U where C where V where R<:Integer}, ::CategoricalArray{S,1,R,V,C,U} where U where C where V) where {S, T, R<:Integer}
Stacktrace:
 [1] MyCustomType(::Array{Int64,1}, ::CategoricalArray{Any,1,UInt32,Any,CategoricalValue{Any,UInt32},Missing}) at ./REPL[164]:2
 [2] construct at /Users/davidlittle/.julia/packages/StructTypes/CpLmq/src/StructTypes.jl:517 [inlined]
 [3] #read#29 at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/structs.jl:450 [inlined]
 [4] read at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/structs.jl:432 [inlined]
 [5] read(::String, ::Type{MyCustomType}; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/structs.jl:17
 [6] read(::String, ::Type{MyCustomType}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/structs.jl:7
 [7] top-level scope at REPL[175]:1  
```

Reply:
Odd... I went and removed these definitions and it now works (even without removing the definition on line 291). I did just switch from 1.4.2 to 1.5 so maybe that was the issue???